catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/IngestCommand.java

```diff
@@ -497,7 +492,13 @@ private void buildQueue(Stream<Path> ingestStream, Queue<Metacard> metacardQueue
                             }
 
                             if (result != null) {
-                                metacardQueue.add(result);
+                                try {
+                                    metacardQueue.put(result);
```

Comment:
I don't see it getting interrupted from within the Ingest Command... So I'm not sure what the behavior would have to be to actually cause this to occur.

Put is a void method so I would have to use something else in ArrayBlockingQueue, like offer with a time limit or the single argument offer with a loop to retry until the operation is successful.